dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationsTask.java

```diff
@@ -0,0 +1,181 @@
+  private static final DateTimeFormatter isoDateTimeFormatter =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+  private final String[] predefinedAppTypes = {
+    "MAPREDUCE",
+  };
+  private final int includedLastDays;
+
+  public ClouderaYarnApplicationsTask(int days) {
+    super(String.format("yarn-applications-%d.jsonl", days));
+    this.includedLastDays = days;
+  }
+
+  @Override
+  protected void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    Preconditions.checkNotNull(
+        clusters, "Clusters must be initialized before fetching YARN applications.");
+
+    List<ApiYARNApplicationDTO> totalYARNApplications = new LinkedList<>();
+    YarnApplicationsLoader appLoader =
+        new YarnApplicationsLoader(handle.getApiURI().toString(), handle.getHttpClient());
+    for (ClouderaClusterDTO cluster : handle.getClusters()) {
+      String clusterName = cluster.getName();
+      List<String> yarnAppTypes = fetchYARNApplicationTypes(handle, clusterName);
+      for (String yarnAppType : yarnAppTypes) {
+        LOG.info(
+            String.format(
+                "Dump YARN applications with %s type from %s cluster", yarnAppType, clusterName));
+        List<ApiYARNApplicationDTO> yarnApplications = appLoader.load(clusterName, yarnAppType);
+        totalYARNApplications.addAll(yarnApplications);
+        LOG.info(
+            String.format(
+                "Dumped %d YARN applications with %s type from %s cluster",
+                yarnApplications.size(), yarnAppType, clusterName));
+      }
+    }
+
+    String yarnAppsJson = parseObjectToJsonString(totalYARNApplications);
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      writer.write(yarnAppsJson);
+    }
+  }
+
+  private String buildISODateTime(int deltaInDays) {
+    ZonedDateTime dateTime =
+        ZonedDateTime.of(LocalDateTime.now().minusDays(deltaInDays), ZoneId.of("UTC"));
+    return dateTime.format(isoDateTimeFormatter);
+  }
+
+  private List<String> fetchYARNApplicationTypes(ClouderaManagerHandle handle, String clusterName) {
+    List<String> yarnApplicationTypes = new ArrayList<>();
+    Collections.addAll(yarnApplicationTypes, predefinedAppTypes);
+
+    String yarnAppTypesUrl =
+        handle.getApiURI().toString() + "clusters/" + clusterName + "/serviceTypes";
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    JsonNode yarnAppTypesJson;
+    ApiYARNApplicationTypeDTO yarnAppTypesFromClouderaAPI;
+    try (CloseableHttpResponse chart = httpClient.execute(new HttpGet(yarnAppTypesUrl))) {
+      yarnAppTypesJson = readJsonTree(chart.getEntity().getContent());
+      yarnAppTypesFromClouderaAPI =
+          parseJsonStringToObject(yarnAppTypesJson.toString(), ApiYARNApplicationTypeDTO.class);
```

Comment:
It seems we do not use `ApiYARNApplicationTypeDTO` anywhere outside this private method. So, we can avoid this class with something like this. 
```
JsonNode at = jsonNode.at("/items");
List<String> strings = objectMapper.convertValue(at, new TypeReference<>() {});
```

Reply:
it's also unclear why the json is 2 times:
1. read `JsonNode` from response
2. convert it to string
3. read object from string - here we also have implicit parsing of string, similar to step #1 

This method should be simplified.

Reply:
Fixed.